lib/state/appActions.js

```diff
@@ -817,4 +822,43 @@ appActions.reset = function(uploadIndex) {
   });
 };
 
+appActions.acknowledge = function (key) {
+  var update = localStore.getItem('acknowledgments');
+
+  if (!update) {
+    update = {};
+  }
+  
+
+  if (!update[key]) {
+    update[key] = {};
+  }
+
+  update[key][this.app.state.targetId] = Date.now();
+  localStore.setItem('acknowledgments', update);
+
+  this.app.setState({
+    acknowledgments: update
+  });
+};
+
+/**
+ * Determine whether the message should be displayed to the user or not
+ *
+ * @param {String} isoString
+ * @return {boolean}
+ */
+appActions.shouldDisplayDaylightSavingsMessage =  function(isoString) {
+  // for now show if the user has not previously acknowledged this warning
+  // in future though, we could make this show if
+  //  - the current date is within X days of DST starting/ending
+  //  - and acknowledged is false or not within X days
+  var state = this.app.state;
+  var localTime = sundial.formatInTimezone(isoString, state.targetTimezone, 'YYYY-MM-DD[T]HH:mm:ss');
```

Comment:
All the gymnastics with `localTime` and then `localTimeTimestamp` are unnecessary. The `localTimeTimestamp` is just `Date.parse(isoString)`. (You're converting from UTC to local and then back again with the current logic.)